Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -113,9 +113,38 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
     /// A class that resolves documentation links by orchestrating calls to other link resolver implementations.
     public var linkResolver = LinkResolver()
     
+    private enum _Provider {
+        case legacy(DocumentationContextDataProvider)
+        case new(DocumentationBundleDataProvider)
+    }
+    private var dataProvider: _Provider
+
     /// The provider of documentation bundles for this context.
-    var dataProvider: DocumentationContextDataProvider
+    var _legacyDataProvider: DocumentationContextDataProvider! {
+        get {
+            switch dataProvider {
+            case .legacy(let legacyDataProvider):
+                legacyDataProvider
+            case .new:
+                nil
+            }
+        }
+        set {
+            dataProvider = .legacy(newValue)
+        }
+    }
     
+    func contentsOfURL(_ url: URL, in bundle: DocumentationBundle) throws -> Data {
+        switch dataProvider {
+        case .legacy(let legacyDataProvider):
+            try legacyDataProvider.contentsOfURL(url, in: bundle)
+        case .new(let dataProvider):
```

Comment:
Should we double check the identifier matches the single bundle's identifier? ...like we do in `bundle(identifier:)`?

Reply:
Yes, since there's a few places that need to keep using this API internally (until after 6.2) it would be good to add an assertion like you suggested in [this comment](https://github.com/swiftlang/swift-docc/pull/1057/files#r1814025232) to ensure that this API is used correctly.

Reply:
Added in https://github.com/swiftlang/swift-docc/pull/1057/commits/96065616aec25b87597cb0a14790d3ea5d9d890d